crates/wash-cli/src/cmd/dev/session.rs

```diff
@@ -394,11 +394,16 @@ impl WashDevSession {
                         .await
                         .context("failed to read line from file")?
                     {
-                        if let Some(host_id) = line
-                            .split_once("host_id=\"")
-                            .map(|(_, rhs)| &rhs[0..rhs.len() - 1])
-                        {
-                            return Ok(host_id.to_string()) as anyhow::Result<String>;
+                        let re = regex::Regex::new(if wasmcloud_opts.enable_structured_logging {
+                            r#"\"host_id\":\s?\"([A-Z0-9]+)\""#
+                        } else {
+                            r#"host_id="([A-Z0-9]+)"#
+                        })
+                        .context("failed to compile regex")?;
+
+                        if let Some(captures) = re.captures(&line) {
+                            return Ok(captures.get(1).unwrap().as_str().to_string())
```

Comment:
Surprised you didn't get asked to use `first()` or `next()` or something -- and it would be nice if we didn't do the `unwrap()` here -- `.context(...)` should work. 

Also, it *feels* like we shouldn't need `as_str().to_string()` -- does the capture not implement `ToString`?

Reply:
I can replace the if with the following, though I'm not sure it's better. The `.get(1)` will only fail if there isn't a capture group, which both patterns have, and the result of `.get()` is `Option<Match>`.
```rs
                        return Ok(re
                            .captures(&line)
                            .context("failed to capture host ID")?
                            .get(1)
                            .context("failed to get capture group")?
                            .as_str()
                            .to_string()) as anyhow::Result<String>;
```

I could also do this, which _should_ be fine since both patterns have at least 1 capture group:
```rs
                        return Ok(
                            re.captures(&line).context("failed to capture host ID")?[1].to_string()
                        ) as anyhow::Result<String>;
```

Any preferences?

Reply:
(I pushed up the latter, but happy to change)

Reply:
Yeah based on what `captures()`'s written contract it should be fine... But I'd probably still do the former even though it's a little bit more verbose just to have graceful error handling there

Reply:
Oh checking now, but you pushed up the *former*, so happy with that!

Reply:
ACK - switched. It'll catch the future case where the regex capture group gets removed.